### PR TITLE
PAccessNetworkInfo can be parsed as a list

### PR DESCRIPTION
**Pull request by xhoaluu**
https://github.com/RestComm/jain-sip/issues/182

https://telestax.zendesk.com/agent/tickets/35976

### CODE REVIEW

**Comment by xhoaluu** [comment]
jenkin result is aligned with master:
https://cxs.restcomm.com/view/VVSSquad/view/JAIN-SIP-Community/job/UPS_JAIN-SIP/93/

File: src/gov/nist/javax/sip/parser/ims/PAccessNetworkInfoParser.java [hunk 1]
```diff
@@ -105,43 +106,53 @@ public SIPHeader parse() throws ParseException
...
+                    break;
+                else
+                    throw createParseException("unexpected char");
             }
-            this.lexer.SPorHT();
-            this.lexer.match('\n');
-
-
-            return accessNetworkInfo;
+            return accessNetworkInfoList;
```
Comment (jaimecasero) [inline src/gov/nist/javax/sip/parser/ims/PAccessNetworkInfoParser.java]: is this going to break existing clients that are using the previous simple PAccessNetworkInfo class?

File: src/test/unit/gov/nist/javax/sip/parser/ims/PAccessNetworkInfoParserTest.java [hunk 2]
```diff
@@ -42,8 +42,10 @@ public void testParser() {
         		"P-Access-Network-Info: 3GPP-UTRAN-TDD; utran-cell-id-3gpp=23456789ABCDE;\"ip=123.123.123.123\"\n",
         		"P-Access-Network-Info: 3GPP-UTRAN-TDD; utran-cell-id-3gpp=23456789ABCDE; [123:4::abcd];rand=l\n",
         		"P-Access-Network-Info: 3GPP-UTRAN-TDD; utran-cell-id-3gpp=23456789ABCDE; [123:4::abcd]\n",
-                "P-Access-Network-Info: IEEE-802.11\n",
-                "P-Access-Network-Info: 3GPP-UTRAN-TDD; utran-cell-id-3gpp=23456789ABCDE\n"
+        		"P-Access-Network-Info: IEEE-802.11\n",
```
Comment (jaimecasero) [inline src/test/unit/gov/nist/javax/sip/parser/ims/PAccessNetworkInfoParserTest.java]: i think this is not really testing multiple occurences of this header to check if list is properly parsed ,right?

**Comment by jaimecasero** [comment]
im not sure if this header must support multiple values or not. apparently there different spec definitions
https://www.ietf.org/mail-archive/web/sipcore/current/msg05804.html

we need to be careful which spec we support here, and what are the impacts for existing apps...

@xhoaluu  have you evaluated this from spec point of view, and JAINSIP spec support?

**Comment by jaimecasero** [comment]
@xhoaluu  i think you need to update this https://github.com/RestComm/jain-sip/blob/master/src/gov/nist/javax/sip/message/ListMap.java with new List type so is regsitered in stack as list compatible

File: src/gov/nist/javax/sip/parser/ims/PAccessNetworkInfoParser.java (hunk 1 above)
Comment (xhoaluu) [reply src/gov/nist/javax/sip/parser/ims/PAccessNetworkInfoParser.java]: I integrated with sip-servlet and test for both single and mutilple PANIs, it's still working.

**Pull request merged**